test/Microsoft.Dnx.Testing.CommonUtils/Directory/Dir.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class Dir
+    {
+        public const string EmptyFile = "";
+        private readonly Dictionary<string, object> _nodes = new Dictionary<string, object>();
+
+        public Dir(string rootPath): this()
+        {
+            Load(rootPath);
+        }
+
+        public Dir()
+        {
+            ReadFile = fileInfo => File.ReadAllText(fileInfo.FullName);
+        }
+
+        public Dir(Dir parent)
+        {
+            ReadFile = parent.ReadFile;
+        }
+
+        public Func<FileInfo, object> ReadFile;
+
+        public string LoadPath { get; private set; } = "In Memory";
+
+        public IEnumerable<KeyValuePair<string, object>> Nodes
+        {
+            get
+            {
+                return _nodes;
+            }
+        }
+
+        public object this[params string[] keys]
```

Comment:
I think it's cleaner to do `["Scripts/bootstrap.js", "Scripts/jquery.js"] = Dir.EmptyFile` for example to create multiple empty files.